### PR TITLE
Tsconfig handling

### LINKED ISSUE #15: Crawler: Handle tsconfig.json while setup
If: Check for base `tsConfigPath` path in  `deprecation-crawler.config.json`. 
- [x] If the file exists take it.
- We have to keep in mind that the process should fail in `interactive` mode false. And we should name it `ci` mode as has to provide special behavior for CI only. 

Else: create separate tsconfig
- [x] Ask for base tsconfig
- [x] List related files
  - [x] Inform on how to change excludes includes.
  - [x] Create `tsconfig.deprecation-crawler.ts`

*Open Questions:*
 Should we commit that change?

### PR DESCRIPTION
**Pull request by BioPhoton**
closes #15

This PR focuses on handling the creation of the ts config file for the crawler. Most important change here, I removed the tsConfigPath from the CrawlerConfig.

### CODE REVIEW

File: packages/deprecation-crawler/src/lib/tasks/ensure-crawler-tsconfig.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import { normalize } from "path";
+import { TSCONFIG_PATH } from "../constants";
+import { findTsConfigFiles } from "../config";
+import { readFileSync, writeFileSync } from "fs";
+import { format, resolveConfig } from "prettier";
+import { fileExists } from "@nrwl/workspace/src/utils/fileutils";
+
+export async function ensureTsConfigPath(config: CrawlConfig): Promise<CrawlConfig> {
+
+  const tsConfigFiles: string[] = findTsConfigFiles();
```
Comment (timdeschryver) [inline packages/deprecation-crawler/src/lib/tasks/ensure-crawler-tsconfig.ts:12]: the `findTsConfigFiles` method can be moved to this file, as its the only place where we need it.

File: packages/deprecation-crawler/src/lib/tasks/ensure-crawler-tsconfig.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  if (tsConfigFiles.length === 0) {
+    throw Error("We need a tsconfig file to crawl");
```
Comment (timdeschryver) [inline packages/deprecation-crawler/src/lib/tasks/ensure-crawler-tsconfig.ts:14]: Do we want to throw here? he could also create the tsconfig file.
Can be done later tho.

File: packages/deprecation-crawler/src/lib/tasks/ensure-crawler-tsconfig.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  }
+
+  if (fileExists(TSCONFIG_PATH)) {
+    // @TODO Should we do any other tests?
```
Comment (timdeschryver) [inline packages/deprecation-crawler/src/lib/tasks/ensure-crawler-tsconfig.ts:18]: I think this is fine for now.

Comment (BioPhoton) [reply packages/deprecation-crawler/src/lib/tasks/ensure-crawler-tsconfig.ts]: Cool! Added info to it for the user.

File: packages/deprecation-crawler/src/lib/tasks/ensure-crawler-tsconfig.ts (hunk 2 above)
Comment (BioPhoton) [reply packages/deprecation-crawler/src/lib/tasks/ensure-crawler-tsconfig.ts]: Refactored to skip this step

File: packages/deprecation-crawler/src/lib/tasks/ensure-crawler-tsconfig.ts (hunk 1 above)
Comment (BioPhoton) [reply packages/deprecation-crawler/src/lib/tasks/ensure-crawler-tsconfig.ts]: moved here. How did you detect that? Manually? Or over ide?

**Comment by BioPhoton** [comment]
On my local machine, the tests fail because everything is one line off. I'm on windows.
Any ideas on that?

**Comment by BioPhoton** [comment]
Resolved conflicts. As this one is quite small let's focus on merging #42 first

**Comment by timdeschryver** [comment]
The failing tests were resolved by the previous PR :)

File: packages/deprecation-crawler/src/lib/tasks/ensure-crawler-tsconfig.ts (hunk 1 above)
Comment (timdeschryver) [reply packages/deprecation-crawler/src/lib/tasks/ensure-crawler-tsconfig.ts]: I just knew it 😅

**Comment by timdeschryver** [comment]
I merged master into this branch and when I tried to run it I ran into several problems:

- it seems like sometimes comments are added to the config files, this throws errors when we parse it to JSON. For example https://github.com/ReactiveX/rxjs/blob/6.5.5/tsconfig/tsconfig.base.json#L13
- I could've selected the wrong config, but it doesn't find deprecations anymore. Maybe we should also copy the `extends` from the original config?

**Comment by BioPhoton** [comment]
I thought about taking the extended once also on account... and skipped it because I'm lazy :D
Will give it a think!

I also thought about the problem of combining it with the --path param over CLI and realized it should also and up in the special place.. But after thinking about it, we could just move the file and edit the tsConfigPath in the config.

**Comment by BioPhoton** [comment]
After our offline discussion I also agree we should skip it. The tsConfigPath is enough. Also, no special name for tsconfig files should be present. We just need to provide cli feedback for further steps

**Comment by BioPhoton** [comment]
closed in favor of #62

**Pull request closed without merging**